core/shared/src/main/scala/org/http4s/package.scala

```diff
@@ -79,4 +82,41 @@ package object http4s {
 
   /** A stream of server-sent events */
   type EventStream[F[_]] = Stream[F, ServerSentEvent]
+
+  // Syntax to enable JS-cross compilation without splitting files
+  // Nobody will actually be able to run these methods b/c they will fail to link
```

Comment:
I'm confused.  These all exist on fs2 and ip4s JVM versions so won't be called.  But they're here for Scala.js which does ... what with them?

Reply:
Scala.js has an extra phase: (1) compile (2) link (3) run. Compiling looks for definitions of the symbols in ordinary bytecode and then linking looks for definitions of the symbols in SJSIR (Scala.js internal representation).

These symbols are missing from the JS versions of fs2 and ip4s, so source fails to compile. By adding these symbols back, we get the code compiling. But then, at the linking step, if somebody has actually tried to use a method that relies on one of these shims, the linker will fail anyway because it cannot find SJSIR for JVMism like `InetSocketAddress`.

So, it's a bit of a hack to avoid file splitting. It would be much better not to offer methods relying on these in the JS version or even at all in favor of the ip4s versions.

Reply:
If we deprecate them, that will prevent the JVM from accidentally calling them, right?  (I think very much that it won't, but just in case.)

Should these be provided in the JS versions of fs2 and ip4s?  They don't have anything to do with http4s?  It's fine as a temporary shim before upstreaming.

Reply:
> If we deprecate them, that will prevent the JVM from accidentally calling them, right?

Sorry, I should clarify: these shims are all `private[http4s]` anyway and we can be confident that they are not sneaking into the JVM versions because our tests would fail in they were.

What I'm advocating we deprecate are the public-facing APIs that _internally_ rely on these shims. For example, any method in `StaticFiles` that uses `nio.Path` instead of the new `fs2.io.Path`. In fact, I've already deprecated those, and we should follow suit for methods using `InetSocketAddress` instead of ip4s `SocketAddress`.

> Should these be provided in the JS versions of fs2 and ip4s?

They shouldn't. Fs2 and ip4s did the right thing here: providing these methods for JS is a lie, because it would allow you to compile code that does not link (and can therefore never ever be run). Ideally, we should do the same (this is how it is currently on main), except that would mean splitting files which means merge pain.

> It's fine as a temporary shim before upstreaming.

Yes, only for 0.23 series. Assuming that we remove all deprecated APIs that rely on these as mentioned above.

Reply:
Okay, that all makes sense.